### DISCUSSION TITLE
Tell us how you use the project!

### DISCUSSION THREAD

**Discussion by nandorholozsnyak** [rodnansol/spring-configuration-property-documenter#74] [Show and tell]
Tell us how you use the project!

- Where did you find the project?
- Why did you start using it?
- Are you maintaining an open source project which utilizes Spring and you have configuration properties?
- Are you using it internally inside your company?
- Which format is your favourite?
- Have you already tried out the custom template option?
- Are you using it with Maven, Gradle or JBang?
- Is there something that you are missing from the project?

**Comment by Riggs333** [comment]
> Where did you find the project?

I was searching via Google for sth. like "Spring Boot configuration properties documentation generation", can not remember what keyword I used exactly. I'm pretty sure along the search I also stumbled upon [this article](https://dev.to/rodnan-sol/spring-configuration-property-documenter-3d9p). 

> Why did you start using it?

In my current project we have a lot of custom properties that need to be documented for admin users of our software. Since I did not want to duplicate and maintain the properties in both places (`@ConfigurationProperties` classes AND some documentation file) this spring-configuration-property-documenter came in handy. :-)
Actually, many years ago I had kind of the same idea for such a tool, but never found the time to fully implement it. 😅 

> Are you maintaining an open source project which utilizes Spring and you have configuration properties?

No, but will spread the word about this tool here to people I know of. ;-)

> Are you using it internally inside your company?

By now, only testing it our in our current customer project.

> Which format is your favourite?

I think, I like asciidoc a lot (at least more than markdown), but HTML looks also quite usable to me.

> Have you already tried out the custom template option?

Not yet.

> Are you using it with Maven, Gradle or JBang?

Maven

> Is there something that you are missing from the project?

I'm just scratching the surface, so, nothing missing here, yet. :-)

**Reply by nandorholozsnyak** [reply]
Thank you for the reply! I appreciate your answers. I hope you like the initiative. Let us know if there is any problem you face.
Issue list is always open for more :D

**Comment by katiaxyz** [comment]
I was searching online for some approach to automatically generate documentation about configuration of Spring-enabled applications. From the search it came out a result with a suggestion to use spring-boot-configuration-processor and the spring-configuration-documenter-plugin.  
I am using asciidoc but also .MD would be good. It would be great also to try out the compact mode.
I haven't tried the custom template option.
I am using the plugin with Maven.
It is not yet clear how to use custom templates.

**Reply by nandorholozsnyak** [reply]
Thank you for your feedback!

**Comment by ParkerM** [comment]
Was originally looking to co-opt some of the Spring Boot tooling that generates those tables with the configprops metadata in appendixes, but lots of manual intervention was sort of required (i.e. creating a list of classes to document, etc) and it would be a maintenance burden continually updating my own bespoke tool if it were based on internal upstream tooling. I think it's matured a bit since the transition to gradle, I believe the last maven one lives in [2.2.x here](https://github.com/spring-projects/spring-boot/tree/2.2.x/spring-boot-project/spring-boot-tools/spring-boot-configuration-docs/src/main/java/org/springframework/boot/configurationdocs), and something similar to it lives in the [changelog generator tool](https://github.com/spring-projects/spring-boot/tree/main/spring-boot-project/spring-boot-tools/spring-boot-configuration-metadata-changelog-generator) now. 

Perhaps they've moved a general generator elsewhere but I abandoned that wild goose chase and nearly wrote my own after some unsuccessful searching before finding this project as a pleasant surprise. I actually forget how I came across this project — either it was mentioned on SO or I saw it while scrolling through a heavily curated GitHub topic filter. Seems very promising and I'm a big fan of the roadmap, saves a lot of time doing meticulous documentation chores before a release.

**Reply by ParkerM** [reply]
Ah, here's where the entrypoint lives for the current gradle impl: https://github.com/spring-projects/spring-boot/blob/96796cc9b508498443751b6c2f10a926a301d49f/buildSrc/src/main/java/org/springframework/boot/build/context/properties/DocumentConfigurationProperties.java

**Reply by nandorholozsnyak** [reply]
Sorry for the late response @ParkerM , but I'm supper happy you use the project.

Thank you for your reply.
